packages/auth/gulpfile.js

```diff
@@ -19,52 +19,82 @@ const closureCompiler = require('gulp-closure-compiler');
 
 // For minified builds, wrap the output so we avoid leaking global variables.
-const OUTPUT_WRAPPER = `(function() {
-  var firebase = require('@firebase/app').default;
-  %output%
-}).call(typeof global !== 'undefined' ? global : typeof self !== 'undefined' ? self : typeof window !== 'undefined' ? window : {});`;
+const CJS_WRAPPER_PREFIX = `(function() {var firebase = require('@firebase/app').default;`;
+const EMS_WRAPPER_PREFIX = `import firebase from '@firebase/app';(function() {`;
+const WRAPPER_SUFFIX = `}).call(typeof global !== 'undefined' ? global : typeof self !== 'undefined' ? self : typeof window !== 'undefined' ? window : {});`;
+
+const wrap = through(function(file) {
+  const makeFile = (prefix, path) =>
+    new File({
+      path,
+      contents: Buffer.concat([
+        Buffer.from(prefix),
+        file.contents,
+        Buffer.from(WRAPPER_SUFFIX)
+      ])
+    });
+
+  this.emit('data', makeFile(CJS_WRAPPER_PREFIX, 'auth.js'));
+  this.emit('data', makeFile(EMS_WRAPPER_PREFIX, 'auth.esm.js'));
+});
 
 // The path to Closure Compiler.
-const COMPILER_PATH = `${path.dirname(require.resolve('google-closure-compiler/package.json'))}/compiler.jar`;
+const COMPILER_PATH = `${path.dirname(
+  require.resolve('google-closure-compiler/package.json')
+)}/compiler.jar`;
 
-const closureLibRoot = path.dirname(require.resolve('google-closure-library/package.json'));
-// Builds the core Firebase-auth JS.
+const closureLibRoot = path.dirname(
+  require.resolve('google-closure-library/package.json')
+);
 
-function buildFirebaseAuth() {
-  return gulp
+// Builds the core Firebase-auth JS.
+const buildFirebaseAuth = () =>
+  gulp
     .src([
       `${closureLibRoot}/closure/goog/**/*.js`,
       `${closureLibRoot}/third_party/closure/goog/**/*.js`,
       'src/**/*.js'
     ])
-    .pipe(closureCompiler({
-      compilerPath: COMPILER_PATH,
-      fileName: 'auth.js',
-      compilerFlags: {
-        closure_entry_point: 'fireauth.exports',
-        compilation_level: OPTIMIZATION_LEVEL,
-        externs: [
-          'externs/externs.js',
-          'externs/grecaptcha.js',
-          'externs/gapi.iframes.js',
-          path.resolve(__dirname, '../firebase/externs/firebase-app-externs.js'),
-          path.resolve(__dirname, '../firebase/externs/firebase-error-externs.js'),
-          path.resolve(__dirname, '../firebase/externs/firebase-app-internal-externs.js')
-        ],
-        language_out: 'ES5',
-        only_closure_dependencies: true,
-        output_wrapper: OUTPUT_WRAPPER
-      }
-    }))
+    .pipe(
+      closureCompiler({
+        compilerPath: COMPILER_PATH,
+        fileName: 'unwrapped.js',
+        compilerFlags: {
+          closure_entry_point: 'fireauth.exports',
+          compilation_level: OPTIMIZATION_LEVEL,
+          externs: [
+            'externs/externs.js',
+            'externs/grecaptcha.js',
+            'externs/gapi.iframes.js',
+            path.resolve(
+              __dirname,
```

Comment:
Both arguments need 2 additional spaces, here and below for each `path.resolve(`.